gf/test/four_index_gf_test.cpp

```diff
@@ -253,6 +253,7 @@ TEST_F(FourIndexGFTest,print)
 TEST_F(FourIndexGFTest, DefaultConstructive)
 {
     gf_type gf_empty;
+#ifndef NDEBUG
```

Comment:
Why to disable the whole test in `NDEBUG` mode?
Actually, i feel that 3 tests are conflated here:
1) Default-constructed GF must be empty;
2) Exception must be thrown if an operation is performed on an empty GF (if `NDEBUG` is not set);
3) Loading is possible into an empty (default-constructed) GF;

At least, the test (2) should be a separate one (and indeed active only if `NDEBUG` is not defined).
This "(1 & 3)" test should be changed:
```c++
    EXPECT_TRUE(gf_empty.is_empty());
    {
        alps::hdf5::archive oar("gf_4i_defconstr.h5","w");
        oar["/gf"] << gf;
    }
    {
        alps::hdf5::archive iar("gf_4i_defconstr.h5");
        iar["/gf"] >> gf_empty;
    }
    EXPECT_FALSE(gf_empty.is_empty());
```